pytorch_lightning/trainer/distrib_data_parallel.py

```diff
@@ -219,6 +220,13 @@ def set_distributed_mode(self, distributed_backend):
             self.use_ddp = True
             self.data_parallel_device_ids = None
             self.on_gpu = False
+        elif distributed_backend == 'horovod':
```

Comment:
it would be nice to be transparent to the user. 
can we automate setting this? this way the abstraction doesn’t bleed?

Reply:
(the mpirun thing)

Reply:
Just to make sure I understand you correctly: is the idea that when running via `horovodrun` or `mpirun`, if the user has *not* specified `distributed_backend`, then we will automatically set `distributed_backend='horovod'` here?

We could certainly do that when running with `horovodrun` + our Gloo backend, as we have special environment variables we can check (`HOROVOD_RANK` for example).  Doing so with `mpirun` is more tricky, because different MPI implementations have different environment variables.  Also, in the future, there might be another distributed backend other than Horovod that uses MPI.

So maybe we could automate it for `horovodrun` but still require them to set it explicitly for `mpirun`?  (Let me know if I misunderstood your suggestion).

Reply:
```
Just to make sure I understand you correctly: is the idea that when running via horovodrun or mpirun, if the user has not specified distributed_backend, then we will automatically set distributed_backend='horovod' here?
```
Yes!

```
So maybe we could automate it for horovodrun but still require them to set it explicitly for mpirun? (Let me know if I misunderstood your suggestion).
```
Let's do this for now (v1) and for v2 maybe we set it explicitely for mpirun? i just don't know enough about mpirun yet, but if mpirun can run any backend then the user should be forced to set it.

Reply:
Sounds good!  I added a `has_horovodrun()` check in `distrib_data_parallel.py` that checks for Gloo or OpenMPI environment variables set by `horovodrun`.  Also added a test.  Let me know if that aligns with what you were thinking.